src/client-side-metrics/exporter.ts

```diff
@@ -0,0 +1,409 @@
+ *   to the Google Cloud Monitoring API. This object contains the project name
+ *   and an array of time series data points, formatted for ingestion by
+ *   Cloud Monitoring.
+ *
+ * @throws Will throw an error if there are issues converting the data.
+ *
+ * @remarks
+ *   The output format is specific to the Cloud Monitoring API and involves
+ *   mapping OpenTelemetry concepts to Cloud Monitoring's data model, including:
+ *   - Mapping resource attributes to resource labels.
+ *   - Mapping metric attributes to metric labels.
+ *   - Handling different metric types (counter, distribution).
+ *   - Converting data points to the correct structure, including start and end
+ *     times, values, and bucket information for distributions.
+ *
+ * @example
+ *   const exportInput: ExportInput = { ... }; // Example ExportInput object
+ *   const monitoringRequest = metricsToRequest(exportInput);
+ *   // monitoringRequest can now be used in monitoringClient.createTimeSeries(monitoringRequest)
+ *
+ *
+ */
+export function metricsToRequest(exportArgs: ExportInput) {
+  const timeSeriesArray = [];
+  const resourceLabels = {
+    cluster: exportArgs.resource._syncAttributes['monitored_resource.cluster'],
+    instance:
+      exportArgs.resource._syncAttributes['monitored_resource.instance_id'],
+    project_id:
+      exportArgs.resource._syncAttributes['monitored_resource.project_id'],
+    table: exportArgs.resource._syncAttributes['monitored_resource.table'],
+    zone: exportArgs.resource._syncAttributes['monitored_resource.zone'],
+  };
+  for (const scopeMetrics of exportArgs.scopeMetrics) {
+    for (const scopeMetric of scopeMetrics.metrics) {
+      const metricName = scopeMetric.descriptor.name;
+      for (const dataPoint of scopeMetric.dataPoints) {
+        const value = dataPoint.value;
+        const allAttributes = dataPoint.attributes;
+        const streaming = allAttributes.streamingOperation;
+        /*
+        metricLabels are built from the open telemetry attributes that are set
+        when a data point is recorded. This means that for one metric there may
+        be multiple time series' with different attributes, but the resource
+        labels will always be the same for a particular export call.
+         */
+        const metricLabels = Object.assign(
+          {
+            app_profile: allAttributes.appProfileId,
+            client_name: allAttributes.clientName,
+            method: allAttributes.methodName,
+            status:
+              (allAttributes as OnAttemptAttribute).attemptStatus?.toString() ??
+              (
+                allAttributes as OnOperationAttribute
+              ).finalOperationStatus?.toString(),
+            client_uid: allAttributes.clientUid,
+          },
+          streaming ? {streaming} : null
+        );
+        const metric = {
+          type: metricName,
+          labels: metricLabels,
```

Comment:
Maybe I'm misunderstanding node, but is converting this to a new metricLabels object necessary? Shouldn't it be equivalent to datapoint.attributes? (assuming the status/streaming attributes were written in the proper format to begin with)

Reply:
In this case it was necessary because the property names upstream were slightly different from the property names here like `client_uid` versus `clientUid`. So I changed the property names upstream so we can just use `datapoint.attributes` because to your point, I think it's nicer if we don't have to do any conversions here.

Reply:
After including more attributes in datapoint.attributes like table, cluster and zone this is now a subset of `datapoint.attributes` so I had to add more code here.